### PR TITLE
issue 22 fix

### PR DESCRIPTION
**Pull request by izhuravlev**
Fixes: #22  
I've added ```get_url_base```, ```get_config_url``` and ```get_config_data``` functions. 
For more information, see issue #22 [comments](https://github.com/vmck/acs-interface/issues/22#issuecomment-538797682)

### CODE REVIEW

File: interface/utils.py [hunk 1]
```diff
@@ -57,3 +47,30 @@ def random_code(length, vocabulary=vocabulary_64):
 def is_true(value):
     text = (value or '').lower().strip()
     return text in ['1', 'yes', 'true', 'on', 'enabled']
+
+    
+def get_url_base(link):
+    m = re.match(r'https://github.com/(?P<org>[^/]+)/(?P<repo>[^/]+)/?$',
+                    link.assignment.repo_url)
```
Comment (Jokeswar) [inline interface/utils.py]: The `link.assignment.repo_url` should start right below `r`. Something like:
```
m = re.match(r'https://github.com/(?P<org>[^/]+)/(?P<repo>[^/]+)/?$',
             link.assignment.repo_url)
```

I suggest using `flake8` as linter. We want the code to respect the `pep8` standard.

File: interface/utils.py [hunk 2]
```diff
@@ -57,3 +47,30 @@ def random_code(length, vocabulary=vocabulary_64):
...
+    url_base = ('https://raw.githubusercontent.com/'
+                '{0}/{1}/'.format(*list(m.groups())))
+    return url_base
+
+def get_config_url(link):
```
Comment (Jokeswar) [inline interface/utils.py]: I think `get_script_url` would be more appropriate.

File: interface/models.py [hunk 3]
```diff
@@ -114,14 +114,7 @@ def state_label(self):
...
     def evaluate(self):
-        m = re.match(r'https://github.com/(?P<org>[^/]+)/(?P<repo>[^/]+)/?$',
-                     self.assignment.repo_url)
-
-        url_base = ('https://raw.githubusercontent.com/'
-                    '{0}/{1}/'.format(*list(m.groups())))
-
-        config_url = urljoin(url_base,
-                             f'{self.assignment.repo_branch}/checker.sh')
+        config_url = get_config_url(self)
```
Comment (Jokeswar) [inline interface/models.py]: `script_url = get_script_url(self)` would be better. Please do not forget to update it a few lines bellow too, at `options['env']['script']`

File: interface/utils.py [hunk 4]
```diff
@@ -57,3 +47,30 @@ def random_code(length, vocabulary=vocabulary_64):
...
+    config_data = requests.get(
+                    urljoin(
+                        url_base,
+                        f'{link.assignment.repo_branch}/config.ini',
+                    )
+                  )
+    return config_data
+
+
+
```
Comment (Jokeswar) [inline interface/utils.py]: Too many new lines at the end of the file. Should be just one, right after `return config_data`. Also, a new line just before `return` would look really nice.

File: interface/utils.py [hunk 5]
```diff
@@ -57,3 +47,30 @@ def random_code(length, vocabulary=vocabulary_64):
...
+                    link.assignment.repo_url)
+    url_base = ('https://raw.githubusercontent.com/'
+                '{0}/{1}/'.format(*list(m.groups())))
+    return url_base
+
+def get_config_url(link):
+    url_base = get_url_base(link)
+    config_url = urljoin(url_base,
+                             f'{link.assignment.repo_branch}/checker.sh')
+    return config_url
```
Comment (Jokeswar) [inline interface/utils.py]: Same thing about the coding style as above applies here. Also, `script_url` is more intuitive than `config_url`.

File: interface/utils.py (hunk 2 above)
Comment (Jokeswar) [inline interface/utils.py]: Two new lines between functions.

File: interface/utils.py [hunk 6]
```diff
@@ -57,3 +47,30 @@ def random_code(length, vocabulary=vocabulary_64):
...
+
+def get_config_data(link):
```
Comment (Jokeswar) [inline interface/utils.py]: Two new lines between functions.

**Comment by izhuravlev** [comment]
Sure, will fix it right away!

**Comment by izhuravlev** [comment]
@Jokeswar made requested corrections

File: interface/utils.py [hunk 7]
```diff
@@ -57,3 +47,32 @@ def random_code(length, vocabulary=vocabulary_64):
...
+    url_base = get_url_base(link)
+    config_data = requests.get(
+                    urljoin(
+                        url_base,
+                        f'{link.assignment.repo_branch}/config.ini',
+                    )
+                  )
+
+    return config_data
```
Comment (Jokeswar) [inline interface/utils.py]: New line at the end of the file. Something like:
```
return config_data

```

**Pull request merged**